collector/__init__.py

```diff
@@ -114,61 +114,51 @@ def __init__(self,
-        self.programs = {}
-
+        self.observations = { site: [] for site in self.sites}
+        self.programs = {site: {} for site in self.sites}
+        self.obs_windows = {site: [] for site in self.sites}
+        self.scheduling_groups = {site: {} for site in self.sites}
         self.night_events = {}
-        self.scheduling_groups = {}
-        self.inst_config = []
 
         # NOTE: This are used to used the EarthLocation and Timezone objects for functions that used those kind of 
         # objects. This can either be include in the Site class if the use of this libraries is justified. 
         self.timezones = {}
         self.locations = {}
 
-        self.programs = {}
-        self.obsid = []
-        self.obstatus = []
-        self.obsclass = []
-        self.band = []
-        self.toostatus = []
-        self.priority = []
-
-        self.tot_time = []  # total program time
-        self.obs_time = []  # used/scheduled time
-        self.obs_windows = []
-
     def load(self, path: str) -> NoReturn:
         """Main collector method. It sets up the collecting process and parameters."""
-        # TODO: Temporary hack for just using one site: this should be a loop over self.sites
-        # TODO: but may require other far-reaching changes (to be evaluated).
-        site_name = Site.GS.value
-
-        xmlselect = [site_name.upper() + '-' + sem + '-' + prog_type
-                     for sem in self.semesters for prog_type in self.program_types]
-
-        # Retrieve the site details
-        try:
-            site = Site(site_name)
-            # TODO: Possibly get rid of these members from Collector and just access them directly?
-            self.locations[site] = GEOGRAPHICAL_LOCATIONS[site]
-            self.timezones[site] = TIME_ZONES[site]
-        except ValueError:
-            raise RuntimeError(f'{site_name} not a valid site name.')
-        except astropy.coordinates.UnknownSiteException:
-            raise RuntimeError(f'{site_name} does not correspond to a valid geographical location.')
-        except pytz.exceptions.UnknownTimeZoneError:
-            raise RuntimeError(f'{site_name} is not associated with a known timezone.')
-
-        self._calculate_night_events()
-
-        # TODO: We will have to modify in order for this code to be usable by other observatories.
-        zip_path = os.path.join(path,
-                                f'{(self.time_range[0] - 1.0 * u.day).strftime("%Y%m%d")}{SITE_ZIP_EXTENSIONS[site]}')
-        logging.info(f'Retrieving program data from: {zip_path}.')
-
-        time_accounting = self._load_tas(path, site)
-        self._readzip(zip_path, xmlselect, site_name, tas=time_accounting, obsclasses=self.obs_classes)
+        
+        
+        for site in self.sites:
+            site_name = site.value
+
+            xmlselect = [site_name.upper() + '-' + sem + '-' + prog_type
+                        for sem in self.semesters for prog_type in self.program_types]
```

Comment:
`program_types` is another place where it seems we should use an `Enum` since there are only so many program types.
@bryanmiller Sorry for the repeated questions, but I assume these are types like Q, C, DD, LP, FT, etc?

Reply:
Sounds good.